src/coreComponents/finiteElement/elementFormulations/H1_Hexahedron_Lagrange1_GaussLegendre2.hpp

```diff
@@ -86,16 +88,69 @@ class H1_Hexahedron_Lagrange1_GaussLegendre2 final : public FiniteElementBase
   virtual ~H1_Hexahedron_Lagrange1_GaussLegendre2() override
   {}
 
+  GEOSX_HOST_DEVICE
   virtual localIndex getNumQuadraturePoints() const override
   {
     return numQuadraturePoints;
   }
 
+  /**
+   * @brief Get the number of quadrature points.
+   * @param stack Stack variables as filled by @ref setupStack.
+   * @return The number of quadrature points.
+   */
+  GEOSX_HOST_DEVICE
+  static localIndex getNumQuadraturePoints( StackVariables const & stack )
+  {
+    GEOSX_UNUSED_VAR( stack );
+    return numQuadraturePoints;
+  }
+
+  GEOSX_HOST_DEVICE
   virtual localIndex getNumSupportPoints() const override
   {
     return numNodes;
   }
 
+  /**
+   * @brief Get the number of support points.
+   * @param stack Object that holds stack variables.
+   * @return The number of support points.
+   */
+  GEOSX_HOST_DEVICE
+  static localIndex getNumSupportPoints( StackVariables const & stack )
+  {
+    GEOSX_UNUSED_VAR( stack );
+    return numNodes;
+  }
+
+  /**
+   * @brief Empty initialization method.
+   * @param nodeManager The node manager.
+   * @param edgeManager The edge manager.
+   * @param faceManager The face manager.
+   * @param cellSubRegion The cell sub-region for which the element has to be initialized.
+   * @param initialization Object that holds initialization properties.
+   */
+  GEOSX_HOST_DEVICE
```

Comment:
Here, if I understood well how the initialization `struct` is going to be used, you may not need this flag either.